pkg/diffs/diffs.go

```
@@ -0,0 +1,136 @@
+				log.Printf("Error reading file %s: %v", path, err)
+			}
+			if bytes.Compare(masterFileContents, prFileContents) != 0 {
+				d.ChangedFiles = append(d.ChangedFiles, path)
+			}
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Diffs) getPresubmitChanges() []string {
+	var preSubmitChangedFiles []string
+	if err := d.findChangedFiles(); err != nil {
+		log.Fatal("Couldn't populate changed files.")
+	}
+	for _, changedFile := range d.ChangedFiles {
+		if strings.Contains(filepath.Base(changedFile), "presubmits.yaml") {
+			preSubmitChangedFiles = append(preSubmitChangedFiles, changedFile)
+		}
+	}
+	return preSubmitChangedFiles
+}
+
+// We return a map of maps here to make the comparing more efficient.
+func getPresubmits(file string) (map[string]map[string]prowconfig.Presubmit, error) {
+	var preSubmits PreSubmits
+
+	preSubmitsMapOfMaps := make(map[string]map[string]prowconfig.Presubmit)
+	pm := make(map[string]prowconfig.Presubmit)
+
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		return preSubmitsMapOfMaps, fmt.Errorf("failed to read presubmit file (%v)", err)
+	}
+
+	if err := yaml.Unmarshal(data, &preSubmits); err != nil {
+		return preSubmitsMapOfMaps, fmt.Errorf("failed to unmarshal presubmit file (%v)", err)
+	}
+
+	for repo, preSubmitList := range preSubmits.PreSubmits {
+		for _, p := range preSubmitList {
+			pm[p.Name] = p
+			preSubmitsMapOfMaps[repo] = pm
+		}
+	}
+	return preSubmitsMapOfMaps, nil
+}
+
+// GetPresubmitsToExecute returns a map of strings with a list of Presubmits to be executed.
+func (d *Diffs) GetPresubmitsToExecute() map[string][]prowconfig.Presubmit {
+	var preSubmitsToExecute []prowconfig.Presubmit
+	preSubmitsToExecuteMap := make(map[string][]prowconfig.Presubmit)
+
+	if preSubmitChangedFiles := d.getPresubmitChanges(); len(preSubmitChangedFiles) > 0 {
+		for _, preSubmitChangedFile := range preSubmitChangedFiles {
+			prPreSubmits, _ := getPresubmits(preSubmitChangedFile)
```

Why ignore errors here?